### PR TITLE
Upgrade to QT 5.15

### LINKED ISSUE #2351: update to Qt 5.15 & then 6

### PR DESCRIPTION
**Pull request by richard-ramos**
Partially fixes #2351 

The UserList as it is implemented right now, causes issues when updating to QT 5.15.2  . Do not merge until StatusQ components to replace this component have been created.

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -15,24 +102,22 @@ RUN export DEBIAN_FRONTEND=noninteractive \
  && sudo add-apt-repository -y ppa:git-core/ppa \
  && sudo apt update -yq \
  && sudo apt install -yq --fix-missing \
-      build-essential cmake git s3cmd libpcre3-dev libnss3 libxcomposite1 libxtst6 jq gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly gstreamer1.0-libav gstreamer1.0-tools gstreamer1.0-alsa gstreamer1.0-pulseaudio
+      build-essential cmake git s3cmd libpcre3-dev libnss3 libxcomposite1 libxtst6 jq \
+      gstreamer1.0-plugins-good gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly \ 
+      gstreamer1.0-libav gstreamer1.0-tools gstreamer1.0-alsa libpulse-mainloop-glib0 \
+      gstreamer1.0-pulseaudio libxrandr2 libxcursor1 libxi6 libxcb-icccm4 libxcb-image0 \
+      libxcb-keysyms1 libxcb-render-util0 libxcb-render0 libxcb-xinerama0
```
Comment (jakubgs) [inline ci/Dockerfile:109]: Wouldn't it make sense to combine all the `apt install commands` into one for one less layer?
Just separate dependencies with comments to indicate what is what.

File: scripts/windows_build_setup.ps1 [hunk 2]
```diff
@@ -72,7 +72,7 @@ export PATH=`"/c/BuildTools/MSBuild/Current/Bin:/c/BuildTools/VC/Tools/MSVC/14.2
 # Stop the script after first error
 $ErrorActionPreference = 'Stop'
 # Version of Qt SDK available form aqt
-$QtVersion = "5.14.2"
+$QtVersion = "5.15.2"
```
Comment (jakubgs) [inline scripts/windows_build_setup.ps1:75]: Was this tested? Does this actually work?

File: ci/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,96 @@
...
+RUN chmod -R 777 /opt
+
+USER user
+WORKDIR /home/user
+ENV HOME /home/user
+
+# Install QT
+RUN python3 -m pip install -U pip
+RUN python3 -m pip install aqtinstall
+RUN python3 -m aqt install --output "$QT_PATH" ${QT_VERSION} linux desktop -m qtwebengine
```
Comment (jakubgs) [inline ci/Dockerfile:90]: Actually, wouldn't it make more sense to cut things off here, and start a new Docker image, and just copy over what `aqt` has installed to the new image without any of the dependencies necessary to do the installation?

https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds

File: ci/Dockerfile (hunk 1 above)
Comment (jakubgs) [reply ci/Dockerfile:109]: If we split the build into two separate images this comment is irrelevant.

File: ci/Dockerfile (hunk 3 above)
Comment (jakubgs) [reply ci/Dockerfile:90]: You can see an example of this in our `nim-waku` repo, but also in many others:
https://github.com/status-im/nim-waku/blob/master/Dockerfile

File: scripts/windows_build_setup.ps1 (hunk 2 above)
Comment (richard-ramos) [reply scripts/windows_build_setup.ps1:75]: The windows build was generated on this PR. But i'm not 100% sure if jenkins did use this branch script?

File: ci/Dockerfile (hunk 3 above)
Comment (richard-ramos) [reply ci/Dockerfile:90]: I'll do that. I did not know that this was something possible to do.

**Review by jrainville — COMMENTED** [review]
QML changes look good 👍

**Comment by PascalPrecht** [comment]
I ran this locally with Qt 5.15.2 and everything seems to work fine!

**Comment by jakubgs** [comment]
I've pushed the Linux image under `statusteam/nim-status-client-build:1.1.0` image I built locally without issues.

Regarding Windows and MacOS, I think we might need to start using different labels depending on what's the QT version:
https://github.com/status-im/status-desktop/blob/34d7279bda5a94b38172b4341213e6dc266f7a07/ci/Jenkinsfile.windows#L4
https://github.com/status-im/status-desktop/blob/34d7279bda5a94b38172b4341213e6dc266f7a07/ci/Jenkinsfile.macos#L4-L6

I think we should start doing what we do for mobile and XCode:
```Jenkinsfile
  agent { label 'macos-xcode-12.5' }
```
This way we can update one of the hosts without affecting the current builds using the previous verison.

**Comment by richard-ramos** [comment]
Replaced by https://github.com/status-im/status-desktop/pull/3499

**Pull request closed without merging**